tools/custom_java_jdbc_client/src/main/java/com/example/BQSimbaExample.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashMap;
+import java.util.Map;
+
+public class BQSimbaExample {
+    private static HashMap<String, String> commandLineArgParser(String[] args) {
+        HashMap<String, String> argMap = new HashMap<>();
+        for (String arg : args){
+            String[] kvPair = arg.split("=");
+            argMap.put(kvPair[0], kvPair[1]);
+        }
+        return argMap;
+    }
+
+    private static String generateConnectionUrl(OAuthUserConfig oAuthConfig, HashMap<String, String> configOptions) {
+        // Build the provided authentication and query options into the JDBC connection URL.
+        StringBuilder connUrlBuilder = new StringBuilder();
+        connUrlBuilder.append("jdbc:bigquery://https://www.googleapis.com/bigquery/v2:443;");
+
+        for (Map.Entry<String, String> userConfig : oAuthConfig.getOAuthUserConfigMap().entrySet()) {
+            connUrlBuilder.append(String.format("%s=%s;", userConfig.getKey(), userConfig.getValue()));
+        }
+        for (Map.Entry<String, String> option : configOptions.entrySet()) {
+            connUrlBuilder.append(String.format("%s=%s;", option.getKey(), option.getValue()));
+        }
+        return connUrlBuilder.toString();
+    }
+
+    public static void main(String[] args) {
+//        HashMap<String, String> argMap = commandLineArgParser(args);  // If args input via command line flags is preferred.
```

Comment:
Uncomment this and add logic to use default values if users don't pass any via cmd line